src/compiler/emitter.ts

```diff
@@ -2660,6 +2673,10 @@ module ts {
                 emit(node.whenFalse);
             }
 
+            function emitGeneratedLabel(node: GeneratedLabel) {
+                write(String(node.labelNumbers[node.label]));

```

Comment:
Using `"" + expr` is fine, but is a bit of an esoteric way to convert a value (much like using `+expr` to convert a value to a number). `expr.toString()` could fail if expr is `undefined`. For clarity I tend to use `String(expr)` (and `Number(expr)`) as, for these intrinsic types, calling the boxed constructors as functions performs a conversion without boxing.

From: Daniel Rosenwasser [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:31 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/emitter.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24627168>:

> @@ -2660,6 +2673,10 @@ module ts {

>                  emit(node.whenFalse);

>              }

>

> +            function emitGeneratedLabel(node: GeneratedLabel) {

> +                write(String(node.labelNumbers[node.label]));

I feel like we tend to use the "" + idiom a bit more, or maybe just .toString().

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24627168>.